src/Data/Key.purs

```diff
@@ -0,0 +1,127 @@
+{-
+Copyright 2015 SlamData, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-}
+
+module Data.Key
+  ( Key()
+  , Platform(..)
+  , alt
+  , character
+  , enter
+  , fromKeyCode
+  , meta
+  , print
+  , printCombination
+  , shift
+  ) where
+
+import Prelude
+import Data.Char (fromCharCode, toUpper)
+import Data.String (fromChar)
+import Data.Foldable (Foldable, foldl)
+
+data Platform = Mac | Other
+
+data Key = Meta | Alt | Shift | Character Char | Enter
+
+meta :: Key
+meta = Meta
+
+alt :: Key
+alt = Alt
+
+shift :: Key
+shift = Shift
+
+character :: Char -> Key
+character = Character <<< toUpper
+
+enter :: Key
+enter = Enter
+
+fromKeyCodeOther :: Int -> Key
+fromKeyCodeOther 13 = Enter
+fromKeyCodeOther 16 = Shift
+fromKeyCodeOther 17 = Meta -- Ctrl
+fromKeyCodeOther 18 = Alt
+fromKeyCodeOther i = Character $ fromCharCode i
+
+fromKeyCodeMac :: Int -> Key
+fromKeyCodeMac 13 = Enter
+fromKeyCodeMac 16 = Shift
+fromKeyCodeMac 91 = Meta -- Left cmd
+fromKeyCodeMac 93 = Meta -- Right cmd
+fromKeyCodeMac 18 = Alt
+fromKeyCodeMac i = Character $ fromCharCode i
+
+fromKeyCode :: Platform -> Int -> Key
+fromKeyCode Mac = fromKeyCodeMac
+fromKeyCode _ = fromKeyCodeOther
+
+printOther :: Key -> String
+printOther Meta = "Ctrl"
+printOther Alt = "Alt"
+printOther Shift = "Shift"
+printOther (Character c) = fromChar c
+printOther Enter = "Enter"
+
+printMac :: Key -> String
+printMac Meta = fromChar $ fromCharCode 8984
+printMac Alt = fromChar $ fromCharCode 8997
+printMac Shift = fromChar $ fromCharCode 8679
+printMac (Character c) = fromChar c
+printMac Enter = "Enter"
+
+print :: Platform -> Key -> String
+print Mac = printMac
+print _ = printOther
+
+seperator :: Platform -> String
+seperator Mac = ""
+seperator _ = "+"
+
+printCombination :: forall f. (Foldable f) => Platform -> f Key -> String
+printCombination p = foldl (printAndAppend p) ""
+  where
+  printAndAppend :: Platform -> String -> Key -> String
+  printAndAppend p s = print p >>> flip append (seperator p) >>> append s
```

Comment:
I think this will print something like `+Ctrl+Alt+S` for the `Other` case, no?

Also, I think this is a tad confusing in point free style (yes, I know this may be surprising coming from me! :smile:) due to strange order of things. Compare:
```
printAndAppend p s k = s <> seperator p <> print p k
```